UnitTests/GitUI.Tests/Editor/FindAndReplaceFormTests.cs

```diff
@@ -0,0 +1,319 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using FluentAssertions;
+using GitUI;
+using ICSharpCode.TextEditor;
+using ICSharpCode.TextEditor.Document;
+using NUnit.Framework;
+
+namespace GitUITests.Editor
+{
+    [Apartment(ApartmentState.STA)]
+    [TestFixture]
+    public class FindAndReplaceFormTests
+    {
+        private FindAndReplaceForm _findAndReplaceForm;
+        private FindAndReplaceForm.TestAccessor _testAccessor;
+        private TextEditorControl _textEditorControl;
+
+        [SetUp]
+        public void SetUp()
+        {
+            _findAndReplaceForm = new FindAndReplaceForm();
+            _textEditorControl = new TextEditorControl();
+            _testAccessor = _findAndReplaceForm.GetTestAccessor();
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+            _findAndReplaceForm.Dispose();
+            _textEditorControl.Dispose();
+        }
+
+        [Test, TestCaseSource(typeof(FindAndReplaceTestData), nameof(FindAndReplaceTestData.MatchCaseCases))]
```

Comment:
Can you please restructure the tests slightly to have the test data defined above each test, similar to how we have it for `TestGetStagedStatus` and `DeleteWithExtremePrejudice_should_return_true_for_empty_input_or_absent_folders`.
This way it is easier to grasp what we're testing and keep the context in your head.

Reply:
Restructured. I also tried to pick better test data so that tests are easier to understand. Please see if it's any better.